scipy/linalg/tests/test_lapack.py

```diff
@@ -1672,23 +1675,39 @@ def test_gttrf_gttrs():
 
         _dl, _d, _du, du2, ipiv, info = dgttrf(dl, d, du)
 
-        np.testing.assert_allclose(dl, diag_cpy[0], atol=atol)
-        np.testing.assert_allclose(d, diag_cpy[1], atol=atol)
-        np.testing.assert_allclose(du, diag_cpy[2], atol=atol)
-        
+        np.testing.assert_array_equal(dl, diag_cpy[0])
+        np.testing.assert_array_equal(d, diag_cpy[1])
+        np.testing.assert_array_equal(du, diag_cpy[2])
+
         b_cpy = b.copy()
         x_dgttrs, info = dgttrs(_dl, _d, _du, du2, ipiv, b)
-        np.testing.assert_allclose(x, x_dgttrs, atol=atol)
-        np.testing.assert_allclose(b, b_cpy, atol=atol)
-        
+        np.testing.assert_array_equal(b, b_cpy)
+
+        np.testing.assert_allclose(x, x_dgttrs, rtol=rtol)
+
+        if index < 2:
+            dl_misshaped = np.random.rand(n)
+            d_misshaped = np.random.rand(n-1)
+            du_misshaped = np.random.rand(n)
+            d_zeros = np.zeros(n)
+        else:
+            d_zeros = np.zeros(n) + np.zeros(n) * 1j
+            dl_misshaped = np.random.rand(n) + np.random.rand(n) * 1j
+            d_misshaped = np.random.rand(n-1) + np.random.rand(n-1) * 1j
+            du_misshaped = np.random.rand(n) + np.random.rand(n) * 1j
 
-        dl_malformatted = np.append(dl, [1])
-        d_malformatted = np.append(d, [1])
-        du_malformatted = np.append(du, [1])
         with assert_raises(ValueError):
-            dgttrf(dl_malformatted, d, du)
+            dgttrf(dl_misshaped, d, du)
         with assert_raises(ValueError):
-            dgttrf(dl, d_malformatted, du)
+            dgttrf(dl, d_misshaped, du)
         with assert_raises(ValueError):
-            dgttrf(dl, d, du_malformatted)
+            dgttrf(dl, d, du_misshaped)
+            
+        singular_matrix_gttrf = dgttrf(dl, d_zeros, du)
+        
+        # np.testing.assert_(singular_matrix_gttrf[5] != 0,
+        #               "?gttrf: singular matrix should return non-zero info.")
+        
+        # put in a 2x2 matrix:    
+        # dgttrf(dl[0], d[:1], du[0])
```

Comment:
I decided to leave these commented out as they will fail if run right now.

Reply:
How do they fail?

Reply:
Oops, should have elaborated more on that. When inputting a singular matrix, it doesn't return non-zero info, so I wrote an assert_() to expect that in the future.  

The 2x2 matrix input does raise an error, but I initially had some trouble getting it to be caught when using `with assert_raises(Exception):`. That's fixed now, thought. Not sure what the issue was but I assume some small typo.

Reply:
> When inputting a singular matrix, it doesn't return non-zero info
What _does_ it do? Does it just output an incorrect solution?

Reply:
Yes.